Code/GraphMol/RGroupDecomposition/RGroupFingerprintScore.cpp

```diff
@@ -0,0 +1,358 @@
+//
+//  Copyright (C) 2020 Gareth Jones, Glysade LLC
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+
+#include "RGroupFingerprintScore.h"
+#include "GraphMol/Fingerprints/Fingerprints.h"
+#include "GraphMol//Fingerprints/MorganFingerprints.h"
+#include "../../../External/GA/util/Util.h"
+#include <memory>
+#include <vector>
+#include <map>
+
+// #define DEBUG
+
+namespace RDKit {
+
+static const int fingerprintSize = 512;
+static const bool useTopologicalFingerprints = false;
+
+// Add fingerprint information to RGroupData
+void addFingerprintToRGroupData(RGroupData *rgroupData) {
+  if (rgroupData->fingerprint == nullptr) {
+    RWMol mol(*rgroupData->combinedMol);
+    for (auto atom : mol.atoms()) {
+      // replace attachment atom by Boron
+      // TODO- Handle multiple attachments differently?
+      if (atom->getAtomicNum() == 0) {
+        atom->setAtomicNum(5);
+        if (atom->getIsotope() > 0) atom->setIsotope(0);
+      }
+    }
+    MolOps::sanitizeMol(mol);
```

Comment:
Will this always sanitize?

Reply:
@bp-kelley I haven't seen it fail, and I can't think of how it would, but I have wrapped it in a try/catch in the latest commit.